test_scripts/Polices/Related_HMI_API/ATF_OnAllowSDLFunctionality_allowed_false_with_device.lua

```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------
+-- Description: 
+--     1. Preconditions: App is registered
+--     2. Steps: Activate App, send SDL.OnAllowSDLFunctionality with 'allowed=false' and with 'device' to HMI
+-- Requirement summary: 
+--     [Policy] "EXTERNAL_PROPRIETARY" flow: Related HMI API
+--     [Policies]: OnAllowSDLFunctionality with 'allowed=false' and with 'device' param from HMI
+--
+-- Expected result:
+--     In case PoliciesManager receives SDL.OnAllowSDLFunctionality with 'allowed=false' and with 'device' param, PoliciesManager must record the named 
+--     device ('device' param) as NOT consented in Local PT ("user_consent_records"-> "device" sub-section) and send BasicCommunication.ActivateApp with 
+--     'level' param of the value from 'default_hmi' key of 'pre-DataConsent'section of Local PT to HMI. App should stay in NONE HMI level
+---------------------------------------------------------------------------------------------
+--[[ General Settings for configuration ]]
+Test = require('connecttest') 
+require('cardinalities')
+require('mobile_session')
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+--[[ Local Functions ]]
+local function get_id_value() 
+  local sql_select = "sqlite3 " .. tostring(SDLStoragePath) .. "policy.sqlite \"SELECT id FROM functional_group WHERE name = "BaseBeforeDataConsent"\""
+    local aHandle = assert( io.popen( sql_select , 'r'))
+    sql_output = aHandle:read( '*l' )   
+    local retvalue = tonumber(sql_output)    
+    if (retvalue == nil) then
+       self:FailTestCase("device id can't be read")
+    else 
+      return retvalue
+    end
+end
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+require('user_modules/AppTypes')
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeleteLogsFileAndPolicyTable()
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:RegisterApp_allowed_false_without_device()
+   -- HMI -> SDL: SDL.ActivateApp
+  local RequestId = self.hmiConnection:SendRequest("SDL.ActivateApp", {appID = self.HMIAppID})
```

Comment:
@vvvakulenko self.HMIAppID is usually assigned at RAI. In the script you are using "connecttest", so you can use:
appID = self.applications[config.application1.registerAppInterfaceParams.appName]